packages/chevrotain/src/parse/cst/cst.ts

```diff
@@ -51,6 +53,52 @@ export function addNoneTerminalToCst(
     } else {
         node.children[ruleName].push(ruleResult)
     }
+
+    setNodeLocation(node, ruleResult.location)
+}
+
+function setNodeLocation(node: CstNode, locationInformation: any) {
+    if (
+        isNaN(node.location.startOffset) ||
+        node.location.startOffset > locationInformation.startOffset
+    ) {
+        node.location.startOffset = locationInformation.startOffset
```

Comment:
I'm not sure about these : as the default value is NaN, we have to test if the current value is NaN before doing the comparison.

Other question : is the information (startOffset, endLine, etc) always present ? It could create some issues if it is missing

Reply:
No its not always present:
https://sap.github.io/chevrotain/documentation/4_3_3/interfaces/ilexerconfig.html#positiontracking

Different levels of position tracking are possible which makes it trickier.
And worst yet we only know this once an input has been provided to the Parser.

In general I would create **four** different CSTPositionAgument methods and assign the relevant
one on the instance during constructor time, so the "correct" one would be in-lined for performance reasons and we would avoid conditional checks on properties that cannot exist.

but in this case we don't even know at constructor time what level of position tracking is needed
unless we provide such an option on the Parser constructor, and we probably should do so anyhow so maybe that is fine.

Reply:
That is what I was thinking... 

Why four CSTPositionAgument methods ? isn't it one for each positionTracking, so 3 ?

Reply:
Oh, I see, you mean 1 for each positionTracking option, and disabling any node location computation ?

Reply:
Yep